src/arch/loong64/Makefile

```diff
@@ -8,6 +8,18 @@ ASM_TCHAR_OPS	:= @
 CFLAGS		+= -fstrength-reduce -fomit-frame-pointer
 CFLAGS		+= -falign-jumps=1 -falign-loops=1 -falign-functions=1
 
+## Check if the follow options is valid for current toolchains
+# -mno-lsx: Disable 128-bit SIMD instructions, added after gcc-14
+# -mno-lasx: Disable 256-bit SIMD instructions, added after gcc-14
+# -mno-relax: Disable linker relaxation, added after gcc-14
+# -Wa,-mno-relax: Disable linker relaxation, added after binutils-2.41
+CHECK_CC_FLAGS := -mno-lsx -mno-lasx -mno-relax -Wa,-mno-relax
+
+define check_cc_flag
+  $(shell $(CC) $(1) -x c -c /dev/null -o /dev/null > /dev/null 2>&1 && echo $(1))
```

Comment:
`-mno-lsx -mno-lasx -mno-relax` are added in the same version of GCC. Maybe they can share one check instead of checking for three times?
btw @mcb30 this looks good to me and does fixed issues on my machine. Could you review this?